src/bp/invoice.rs

```diff
@@ -0,0 +1,128 @@
+// LNP/BP Core Library implementing LNPBP specifications & standards
+// Written in 2020 by
+//     Dr. Maxim Orlovsky <orlovsky@pandoracore.com>
+//
+// To the extent possible under law, the author(s) have dedicated all
+// copyright and related and neighboring rights to this software to
+// the public domain worldwide. This software is distributed without
+// any warranty.
+//
+// You should have received a copy of the MIT License
+// along with this software.
+// If not, see <https://opensource.org/licenses/MIT>.
+
+use bitcoin::hashes::sha256d;
+use bitcoin::secp256k1::Signature;
+use bitcoin::Address;
+use miniscript::{descriptor::DescriptorPublicKey, Descriptor, Miniscript};
+use url::Url;
+
+use crate::bp::blind::OutpointHash;
+use crate::bp::chain::AssetId;
+use crate::bp::{Chain, HashLock, P2pNetworkId, Psbt, ScriptPubkeyFormat};
+use crate::lnp::{tlv, Features};
+use crate::secp256k1;
+
+#[derive(Tlv)]
+#[lnpbp_crate(crate)]
+pub enum FieldType {
+    #[tlv(type = 0x01)]
+    Payers,
+}
+
+#[derive(Lnp)]
+#[tlv_types(FieldType)]
+#[lnpbp_crate(crate)]
+pub struct Invoice {
```

Comment:
In general this struct can grow to arbitrary size and, being so generic, it's probably difficult to identify a very compact encoding for it. This might make it impossible to pass it via NFC or QRcode which currently are the preferred methods for invoice sharing. Does this have a different usecase or am I missing something?

Reply:
So is this going to be bolt-11 compliant for invoices that include lightning-only information?

Reply:
This is unrelated. Two unrelated stories. The first one is that this standard uses the same encoding as LN. It does not mean it makes it interoperable, it's just a timeproven way. And also this simplifies conversion of this new invoice type into a lightning message if one day someone will need that (which can't be done with current BOLT-11 by the way).

The second story is that the proposed invoice is parsable from BOLT-11 invoice or can be converted into BOLT-11 invoice. But this has nothing to do with used TLV mechanism